release notes/v0.31.0.md

```diff
@@ -0,0 +1,75 @@
+  { expected_response:true }...: avg=203.31ms min=203.31ms med=203.31ms max=203.31ms p(90)=203.31ms p(95)=203.31ms
+http_req_failed................: 50.00% ✓ 1   ✗ 1
+```
+
+Note the new `http_req_duration` sub-metric for passed requests only and the `http_req_failed` Rate metric.
+
+The `http.setResponseCallback()` is meant to allow arbitrary JS functions to process responses, but for now only the `http.expectedStatuses()` callback is supported.
+
+
+## Other enhancements and UX improvements
+
+- JS: Because of the awesome improvements to [goja](https://github.com/dop251/goja), the JS runtime k6 uses, it's no longer necessary for k6 to load [core.js](https://github.com/zloirock/core-js) to polyfill missing JS features when using the default [`--compatibility-mode=extended`](https://k6.io/docs/using-k6/javascript-compatibility-mode). So in v0.31.0 core.js has been dropped entirely, yielding some significant CPU and memory usage improvements. The actual numbers will depend on the use case, but users can expect [TODO: Mention a percentage range?] ([#1824](https://github.com/loadimpact/k6/pull/1824))
+- JS: Also because of goja improvements, some Babel plugins were disabled which should decrease CPU and memory usage even more. ([#1822](https://github.com/loadimpact/k6/pull/1822))
+- JS: Expanded `ArrayBuffer` support in most internal modules, so now you can pass `ArrayBuffer` to `http.file()`, in `k6/encoding` and `k6/crypto` functions. This makes working with binary files more efficient as it doesn't require string translations. In upcoming versions we plan to expand this to the WebSocket module, as well as make some potentially breaking changes for APIs that currently return string (see the details in the Breaking Changes announcement below). ([#1800](https://github.com/loadimpact/k6/pull/1800))
+- The Docker image base was updated to Alpine 3.13. Thanks @andriisoldatenko! ([#1821](https://github.com/loadimpact/k6/pull/1821))
+- The Debian package now includes `ca-certificates` as a dependency. Thanks @Bablzz! ([#1854](https://github.com/loadimpact/k6/pull/1854))
+
+
+## Bugs fixed!
+
+- JS: Babel is now also run on compilation errors, which improves support of some obscure language features. ([#1861](https://github.com/loadimpact/k6/pull/1861))
+
+
+## Internals
+
+- JS: Some TC39 tests for unsupported features were disabled, improving the runtime of the test suite. ([#1816](https://github.com/loadimpact/k6/pull/1816))
+- CI: Some more tests were enabled on Windows. ([#1855](https://github.com/loadimpact/k6/pull/1855))
+
+
+## Breaking changes
+
+- JS: While we don't expect the core.js removal and Babel changes to impact the vast majority of users, those were substantial changes in how k6 interprets JS and a minority of users might experience issues with their tests. Please report any unexpected JavaScript errors by creating a GitHub issue.
```

Comment:
We should probably mention that [this](https://stackoverflow.com/a/65849645/5427244) hack will no longer work due to the dropping of Promise and some babel plugins

Reply:
Well, it won't work as is, but they'd have to add more Babel plugins to the list?

Reply:
and adding a Promise polyfill, probably, I haven't actually tried that workaround ever so :shrug: .

Reply:
given that at least one person has told us they made it work and you have proposed it as a solution, I would argue we should at least say that we specifically drop some stuff that made it work and now users will need to add them back if they want to use that hack.

Reply:
OK, see e02862f6.